### ISSUE TITLE
Hot Reload for tests

### ISSUE THREAD

**Issue by passsy** [flutter/flutter#25973]
Hot Reload is one of the biggest selling points of Flutter. It makes development fast. At least when writing user interfaces. Unit tests don't support Hot Reload. With Dart2, spawning a new Isolate for each test file, tests became rather slow (multiple seconds until seeing first results).

It would be great when Hot Reload would also work for tests, making not only UI development fast but also TDD fun.

Related https://github.com/dart-lang/sdk/issues/32953

**Comment by passsy** [comment]
Good idea but it doesn't help. Here's an example with 12 basic unit tests in 4 files

```
> time flutter test
00:02 +12: All tests passed!
flutter test  5.22s user 1.04s system 157% cpu 3.983 total

> time flutter test -j1
00:03 +12: All tests passed!
flutter test -j1  5.23s user 1.04s system 146% cpu 4.280 total
```

The execution time of all tests is `100ms` (Shown in IntelliJ). But the whole startup procedure, required for each test run, add an overhead of `5s`.
I wish there would be something like `pub webdev serve` but for tests, having a running Isolate daemon waiting for the next test execution.

**Comment by zoechi** [comment]
Similar to #17130, #4719

**Comment by matanlurey** [comment]
This is not planned (in particular, for Dart2JS, there is no such thing as hot reload).
